test/support/case.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Alice.Handlers.Case do
+  def all_replies() do
+    message = receive do
+      {:send_message, %{response: message}} -> message
+    after
+      0 -> nil
+    end
+    case message do
+      nil -> []
+      message -> [message | all_replies()]
+    end
+  end
+
+  def first_reply() do
+    case all_replies() do
+      [first_message |  _] -> first_message
+      _                    -> nil
+    end
+  end
+
+  def fake_conn(), do: fake_conn("")
+  def fake_conn(text) do
+    %Alice.Conn{message: %{text: text, channel: :channel, user: :fake_user}, slack: %{users: [fake_user: %{name: "fake_user"}], me: %{id: :alice}}}
+  end
+
+  def fake_conn_with_capture(message, capture_regex) do
+    fake_conn(message)
+    |> Alice.Conn.add_captures(capture_regex)
+  end
+
+  def receive_message(message) do
+    conn = fake_conn(message)
+    case Alice.Conn.command?(conn) do
+      true  -> Alice.Router.match_commands(conn)
+      false -> Alice.Router.match_routes(conn)
+    end
+  end
```

Comment:
I was wrong. We should call this `send_message`. "Receive" usually implies some kind of callback in Elixir, and we're sending the message to the bot from the test.

Reply:
Also, I think we're going to need a version of this function that takes a `conn`. I can definitely imagine some handler tests that need to build up a conn with some unique data.
```suggestion
  def send_message(%Alice.Conn{} = conn) do
    case Alice.Conn.command?(conn) do
      true -> Alice.Router.match_commands(conn)
      false -> Alice.Router.match_routes(conn)
    end
  end
  
  def send_message(message) do
    message
    |> fake_conn()
    |> send_message()
  end
```